packages/contracts-periphery/contracts/universal/faucet/Faucet.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { IFaucetAuthModule } from "./authmodules/IFaucetAuthModule.sol";
+
+/**
+ * @title  SafeSend
+ * @notice Sends ETH to a recipient account without triggering any code.
+ */
+contract SafeSend {
+    /**
+     * @param _recipient Account to send ETH to.
+     */
+    constructor(address payable _recipient) payable {
+        selfdestruct(_recipient);
+    }
+}
+
+/**
+ * @title  Faucet
+ * @notice Faucet contract that drips ETH to users.
+ */
+contract Faucet {
```

Comment:
Does this contract also need to extend `EIP712Upgradeable` in order for it to be upgradeable in the future?

Reply:
I don't think we really need to worry about upgradeability here. It's easy enough to claw back the funds and deploy a new faucet.